File: MS.NET/Fundamentals/Generic/Basics/genmethodtest.cs
using System;

static class Program
{
	private static T Select<T>(int selector, T first, T second)
	{
		if(selector < 0)
			return first;
		return second;
	}

	public static void Main(string[] args)
	{
		int s = args.Length > 0 ? int.Parse(args[0]) : 0;

		string ss = Select(s, "Monday", "Tuesday");
		Console.WriteLine("Selected string = {0}", ss);

		double sd = Select(s, 7.25, 2.75);
		Console.WriteLine("Selected double = {0}", sd);

		int si = Select(s, 123456, 0xabcdef);
		Console.WriteLine("Selected int = {0}", si);
	}
}

File: Java/Fundamentals/Generics/Basics/GenMethodTest.java
class GenMethodTest{

	private static<T> T select(int selector, T first, T second){
		if(selector < 0)
			return first;
		return second;
	}

	public static void main(String[] args){
		int s = Integer.parseInt(args[0]);
		String ss = select(s, "Monday", "Tuesday");
		System.out.printf("Selected String = %s%n", ss);
		double sd = select(s, 4.3, 3.2);
		System.out.printf("Selected double = %s%n", sd);
		int si = select(s, 123456, 0xABCDEF);
		System.out.printf("Selected int = %s%n", si);
	}
}


